.github/workflows/changelog-check.yml

```diff
@@ -0,0 +1,59 @@
+name: CHANGELOG Unreleased Section
+
+"on":
+  push: {branches: [main]}
+
+jobs:
+  check-changelog:
+    runs-on: ubuntu-latest
+    name: Verify CHANGELOG.md has unreleased entries
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v5
+
+      - name: Extract and validate Unreleased section
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+
+            // Read CHANGELOG.md
+            const changelog = fs.readFileSync('CHANGELOG.md', 'utf8');
+
+            // Split into lines
+            const lines = changelog.split('\n');
+
+            // Find the "## Unreleased" section
+            const unreleasedIndex = lines.findIndex(line => line.trim() === '## Unreleased');
+
+            if (unreleasedIndex === -1) {
+              core.setFailed('❌ ERROR: Could not find "## Unreleased" section in CHANGELOG.md');
+              return;
+            }
+
+            // Find the next "## " section (next version)
+            const nextSectionIndex = lines.findIndex((line, index) =>
+              index > unreleasedIndex && line.match(/^## \d+\.\d+\.\d+/)
+            );
+
+            // Extract content between sections
+            const endIndex = nextSectionIndex === -1 ? lines.length : nextSectionIndex;
+            const unreleasedLines = lines.slice(unreleasedIndex + 1, endIndex);
+
+            // Filter out empty lines and whitespace-only lines
+            const contentLines = unreleasedLines.filter(line =>
+              line.trim().length > 0
+            );
+
+            console.log('Extracted unreleased content:');
+            contentLines.forEach(line => console.log(line));
+            console.log('---');
+
+            // Check if there's any actual content
+            if (contentLines.length === 0) {
+              core.setFailed('❌ ERROR: CHANGELOG.md "## Unreleased" section is empty or contains only blank lines\nPlease add at least one entry describing the changes in this release.');
+            } else {
+              console.log(`✅ SUCCESS: CHANGELOG.md "## Unreleased" section contains entries`);
+              console.log(`Found ${contentLines.length} non-blank lines`);
+            }
```

Comment:
Would we just create our own action that uses this code? I don't imagine we'd have to import this whole code block on every cookbook.

Reply:
Yeah referencing this block is probably the quickest & simplest thing.